engine/src/main/java/io/zeebe/engine/processing/incident/ResolveIncidentProcessor.java

```diff
@@ -97,28 +104,63 @@ private void attemptToResolveIncident(
       // todo (#6000): resolve incident for UNHANDLED_ERROR_EVENT
       // at time of writing the workflow cannot be fixed, so this incident cannot be resolved
     } else {
-      attemptToContinueProcessProcessing(responseWriter, streamWriter, sideEffect, incidentRecord);
+      attemptToContinueWorkflowProcessing(
+          command, responseWriter, streamWriter, sideEffect, incidentRecord);
     }
   }
 
-  private void attemptToContinueProcessProcessing(
+  private void attemptToContinueWorkflowProcessing(
+      final TypedRecord<IncidentRecord> command,
       final TypedResponseWriter responseWriter,
       final TypedStreamWriter streamWriter,
       final Consumer<SideEffectProducer> sideEffect,
       final IncidentRecord incidentRecord) {
-    final long elementInstanceKey = incidentRecord.getElementInstanceKey();
-    final IndexedRecord failedRecord = elementInstanceState.getFailedRecord(elementInstanceKey);
 
-    if (failedRecord != null) {
+    getFailedCommand(incidentRecord)
+        .ifRightOrLeft(
+            failedCommand -> {
+              sideEffects.clear();
+              sideEffects.add(responseWriter::flush);
 
-      sideEffects.clear();
-      sideEffects.add(responseWriter::flush);
+              bpmnStreamProcessor.processRecord(
+                  failedCommand, noopResponseWriter, streamWriter, sideEffects::add);
```

Comment:
Wouldn't this also mean that we can only migrated that if all bpmn stream processors which cause incidents are migrated?

Reply:
No, this is only executed on processing `Incident:Resolve`, so it is not reprocessed.

Reply:
Ah because you already marked it as migrated right?

Reply:
Yup, Incident records will be migrated, if on processing the resolve we produce records that have not been migrated yet, then those records are reprocessed as always